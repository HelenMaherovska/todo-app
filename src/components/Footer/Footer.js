import React from "react";
import { FILTERS } from "../../constants";
import FilterBtn from "../../components/FilterBtn/FilterBtn";

const VisibleTodoList = () => {
  return Object.keys(FILTERS).map(filterBtn => (
    <FilterBtn key={FILTERS[filterBtn]} filter={FILTERS[filterBtn]}>
      {FILTERS[filterBtn]}
    </FilterBtn>
  ));
};

export default VisibleTodoList;
