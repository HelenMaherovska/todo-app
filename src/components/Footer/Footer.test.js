import React from "react";
import { shallow } from "enzyme";
import Footer from "./Footer";

describe("Footer", () => {
  const footer = shallow(<Footer />);

  it("renders correctly", () => {
    expect(footer).toMatchSnapshot();
  });

  it("creates a FilterBtn component", () => {
    expect(footer.find("Connect(FilterBtn)").exists()).toBe(true);
  });

  it("creates a list of FilterBtn component", () => {
    expect(footer.find("Connect(FilterBtn)")).toHaveLength(3);
  });
});
