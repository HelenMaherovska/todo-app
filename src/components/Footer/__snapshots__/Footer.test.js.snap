// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VisibleTodoList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "ALL",
    "nodeType": "class",
    "props": Object {
      "children": "ALL",
      "filter": "ALL",
    },
    "ref": null,
    "rendered": "ALL",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "ALL",
      "nodeType": "class",
      "props": Object {
        "children": "ALL",
        "filter": "ALL",
      },
      "ref": null,
      "rendered": "ALL",
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "COMPLETE",
      "nodeType": "class",
      "props": Object {
        "children": "COMPLETE",
        "filter": "COMPLETE",
      },
      "ref": null,
      "rendered": "COMPLETE",
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "ACTIVE",
      "nodeType": "class",
      "props": Object {
        "children": "ACTIVE",
        "filter": "ACTIVE",
      },
      "ref": null,
      "rendered": "ACTIVE",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
